lib/tesla/middleware/core.ex

```diff
@@ -125,6 +125,10 @@ defmodule Tesla.Middleware.Opts do
   @behaviour Tesla.Middleware
 
   @impl Tesla.Middleware
+  def call(env, next, adapter: adapter_opts) do
```

Comment:
This pattern will only match `[adapter: x]` or `[adapter: x, other: opt]` but not `[other: ops, adapter: x]`.